server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectConsoleService.java

```diff
@@ -401,48 +329,19 @@ public SqlAsyncExecuteResp streamExecute(@NotNull String sessionId,
-        PreConditions.validArgumentState(Objects.nonNull(requestId), ErrorCodes.SqlRegulationRuleBlocked, null, null);
-        ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId);
-        Future<List<JdbcGeneralResult>> listFuture =
-                ConnectionSessionUtil.getFutureJdbcResult(connectionSession, requestId);
-        int timeout = Objects.isNull(timeoutSeconds) ? DEFAULT_GET_RESULT_TIMEOUT_SECONDS : timeoutSeconds;
-        try {
-            List<JdbcGeneralResult> resultList = listFuture.get(timeout, TimeUnit.SECONDS);
-            Map<String, Object> context = ConnectionSessionUtil.getFutureJdbcContext(connectionSession, requestId);
-            ConnectionSessionUtil.removeFutureJdbc(connectionSession, requestId);
-            return resultList.stream().map(jdbcGeneralResult -> {
-                Map<String, Object> ctx = context == null ? new HashMap<>() : context;
-                SqlExecuteResult result = generateResult(connectionSession, jdbcGeneralResult, ctx);
-                try (TraceStage stage = result.getSqlTuple().getSqlWatch().start(SqlExecuteStages.SQL_AFTER_CHECK)) {
-                    sqlInterceptService.afterCompletion(result, connectionSession, new AsyncExecuteContext(null, ctx));
-                } catch (Exception e) {
-                    throw new IllegalStateException(e);
-                }
-                return result;
-            }).collect(Collectors.toList());
-        } catch (InterruptedException | ExecutionException e) {
-            throw new IllegalStateException(e);
-        } catch (TimeoutException timeoutException) {
-            if (log.isDebugEnabled()) {
-                log.debug("Get sql execution result timed out, sessionId={}, requestId={}", sessionId, requestId,
-                        timeoutException);
-            }
-            return Collections.emptyList();
-        }
+    public AsyncExecuteResultResp getMoreResults(@NotNull String sessionId, String requestId) {
+        return getMoreResults(sessionId, requestId, null);
     }
 
-    public AsyncExecuteResultResp getMoreResults(@NotNull String sessionId, String requestId) {
+    public AsyncExecuteResultResp getMoreResults(@NotNull String sessionId, String requestId, Integer timeoutSeconds) {
         PreConditions.validArgumentState(Objects.nonNull(requestId), ErrorCodes.SqlRegulationRuleBlocked, null, null);
         ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId);
         AsyncExecuteContext context =
                 (AsyncExecuteContext) ConnectionSessionUtil.getExecuteContext(connectionSession, requestId);
-        boolean shouldRemoveContext = context.isFinished();
+        int timeout = Objects.isNull(timeoutSeconds) ? DEFAULT_GET_RESULT_TIMEOUT_SECONDS : timeoutSeconds;
+        boolean shouldRemoveContext = false;
         try {
+            shouldRemoveContext = await(context, timeout, TimeUnit.SECONDS);
```

Comment:
I think that we should not do this, assume that I get two sqls:
1. select 1 from dual;
2. select sleep(200) from dual;

the first one needs less time to execute but we await for both two sql executed completely for 1 second, It makes nonsense~

Reply:
done. We would use a more reasonable logic here. 
If the results are not empty, ODC will return them instantly. Otherwise, it will busy-waiting for 1s to get any result.